r/R/query-engine.R

```diff
@@ -114,11 +112,13 @@ ExecPlan <- R6Class("ExecPlan",
             x
           })
         }
+        target_names <- names(.data$aggregations)
+        for (i in seq_len(length(target_names))) {
+          .data$aggregations[[i]][["name"]] <- .data$aggregations[[i]][["target"]] <- target_names[i]
+        }
```

Comment:
Maybe using [`purrr::imap()`](https://purrr.tidyverse.org/reference/imap.html) here will make the lint warning go away? I don't see how this current version is more complex than what is currently on master.

```suggestion
        .data$aggregations <- imap(.data$aggregations, function(x, name) {
          # Embed the name inside the aggregation objects
          # `target` and `name` are the same because we Project the data that way
          x[["name"]] <- x[["target"]] <- name
          x
        })
```

Reply:
I understand, let's see how it goes in the CIs.